operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -149,21 +148,6 @@ public void stop() throws OperatorException {
         watchedNS);
   }
 
-  /**
-   * Method to register operator and facilitate configuration override.
-   *
-   * @param reconciler part of the reconciler to register
-   * @param configOverrider consumer to use to change config values
-   * @param <R> the {@code HasMetadata} type associated with the reconciler
-   */
-  public <R extends HasMetadata> void register(Reconciler<R> reconciler,
-      Consumer<ControllerConfigurationOverrider> configOverrider) {
-    final var controllerConfiguration = configurationService.getConfigurationFor(reconciler);
```

Comment:
The default is to add AnnotationControllerConfiguration, see:
http://github.com/java-operator-sdk/java-operator-sdk/blob/dbf6c2e001ff5bba701f231babc9affc8c7433a7/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/DefaultConfigurationService.java#L33-L33

Reply:
Like the other changes, but have basically conceptual problems with this, using `DependentResourceConfigurationProvider` makes the whole Annotation based Dependent Resource definition kinda loosing it's meaning. 
So from now in the background what will happen is just the DependentResource will be instantiated (with constaint of empty constructor present), and the the configuration that is explicitly implemented in the reconciler is passed to it. 

I really liked that the configuration of the dependent resources is decoupled from the reconciler, since it makes then the reconciler much cleaner. Now rather I would then do just standalone DependentResources instead of this, when implementing an operator. Since we loose even that decoupling.

(And of course the reconcile and prepare event sources will be called explicitly in the background but that is again not a huge benefit and rather could be confusing / hard to understand for people who start with. So has negative sides too)

So I think we should either:

1. Fix the configuraiton with using ControllerConfiguration mechanism
2. Or just drop the whole annotation based DependentResource definitions (/managed dependent resources) since it is starts loosing it's benefit, and seems to be too hard and even confusing to implement / work with it.

Reply:
> The default is to add AnnotationControllerConfiguration, see: [http://github.com/java-operator-sdk/java-operator-sdk/blob/dbf6c2e001ff5bba701f231babc9affc8c7433a7/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/DefaultConfigurationService.java#L33-L33](https://github.com/java-operator-sdk/java-operator-sdk/blob/dbf6c2e001ff5bba701f231babc9affc8c7433a7/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/DefaultConfigurationService.java#L33-L33)

checked, this works no on `next` branch.

Reply:
An alternative what a I see is to pass this on registration:
`operator.register(controller, cofiguration, dependentResourceConfiguration)`  and
`operator.register(controller, dependentResourceConfiguration)`

So decouple it from controller configuration.